test-runner-poc/config.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+const f = require('util').format;
+const url = require('url');
+const qs = require('querystring');
+const core = require('../lib/core');
+class NativeConfiguration {
+  constructor(environment) {
+    this.options = environment || {};
+    this.host = environment.host || 'localhost';
+    this.port = environment.port || 27017;
+    this.db = environment.db || 'integration_tests';
+    this.mongo = environment.mongo;
+    this.setName = environment.setName || 'rs';
+    this.require = this.mongo;
+    this.writeConcern = function() {
+      return { w: 1 };
+    };
+    this.topology = environment.topology || this.defaultTopology;
+    this.environment = environment;
+    if (environment.setName) {
+      this.replicasetName = environment.setName || 'rs';
```

Comment:
Do we need both `replicasetName` and `setName`?

Reply:
Yes - `configuration.setName` and `configuration.replicasetName` are both looked up by tests.